### PR TITLE
Split logic routines from math

### PR DESCRIPTION
**Pull request by emcastillo**
Move the functions of IsNan, IsFinite, etc from the math.{h,cc} routine to logic.{h,cc} 
The reason to move them to logic is because of the way they are structured in numpy 

https://docs.scipy.org/doc/numpy/reference/routines.logic.html

However, in cuda kernels this funcions are located in misc.cu, and same for the native device. and there is no logic.cu file but comparisson one.

Should we move them to a logic.cu in order to keep consistency?

### CODE REVIEW

**Comment by niboshi** [comment]
@emcastillo Yes, please move them for consistency.

File: chainerx_cc/chainerx/routines/math.h [hunk 1]
```diff
@@ -13,14 +13,4 @@ Array Sigmoid(const Array& x);
...
 
-Array IsNan(const Array& x);
-
-Array IsInf(const Array& x);
-
-Array IsFinite(const Array& x);
-
-Array Ceil(const Array& x);
-
-Array Floor(const Array& x);
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/math.h]: Ops `Floor` and `Ceil` should not be moved no?

File: chainerx_cc/chainerx/routines/logic.h [hunk 2]
```diff
@@ -56,4 +56,14 @@ Array All(const Array& a, const OptionalAxes& axis = nonstd::nullopt, bool keepd
...
 
+Array IsNan(const Array& x);
+
+Array IsInf(const Array& x);
+
+Array IsFinite(const Array& x);
+
+Array Ceil(const Array& x);
+
+Array Floor(const Array& x);
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/logic.h]: Ditto.

**Comment by hvy** [comment]
Basically LGTM, left some smaller comment.

File: chainerx_cc/chainerx/routines/logic.h (hunk 2 above)
Comment (emcastillo) [reply chainerx_cc/chainerx/routines/logic.h]: There is another PR tackling this #7407

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins failures seem unrelated.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**